src/classes/PSC_Opportunity_TDTM.cls

```diff
@@ -63,29 +65,39 @@ public with sharing class PSC_Opportunity_TDTM extends TDTM_Runnable {
     * @param listOld The list of opportunities from trigger old. 
-    * @return list<Partial_Soft_Credit__c> Partial soft credits with modified currency that need updating via DML.
+    * @return list<sObject> Partial Soft Credits or Account Soft Credits with modified currency that need updating via DML.
     ********************************************************************************************************/
-    public static list<Partial_Soft_Credit__c> updatePartialSoftCreditCurrencyFromParent(list<sObject> newlist, list<sObject> oldlist) {
-        list<Partial_Soft_Credit__c> results = new list<Partial_Soft_Credit__c>();
+    public static list<SObject> updateSoftCreditCurrencyFromParent(list<SObject> newlist, list<SObject> oldlist, SObjectType objType) {
+        list<SObject> results = new List<SObject>();
         //map holds object id to currency iso code
-        map<id,string> mapOppIdToISO = new map<id,string>();
+        Map<Id,String> mapOppIdToISO = new Map<Id,String>();
 
         //if the parent currency has changed, add to map
-        for (integer i=0;i<newlist.size();i++) {
+        for (Integer i=0;i<newlist.size();i++) {
             if (newlist[i].get('CurrencyIsoCode')!=oldlist[i].get('CurrencyIsoCode')) {
                 mapOppIdToISO.put((id)newlist[i].get('id'),(string)newlist[i].get('CurrencyIsoCode'));
             }
         }
 
         //can't put mapOppIdToISO.keySet() in dynamic soql query, so creating new set to hold the IDs
-        set<id> oppIds = mapOppIdToISO.keySet();
+        Set<Id> oppIds = mapOppIdToISO.keySet();
+
+        if (objType == Partial_Soft_Credit__c.SObjectType) {
+            //query all PSCs associated with these opps, update their ISO code
+            for (Partial_Soft_Credit__c psc : database.query('SELECT Id, Opportunity__c, CurrencyIsoCode FROM Partial_Soft_Credit__c WHERE Opportunity__c IN :oppIds')) {
+                string oppCurrency = mapOppIdToISO.get(psc.Opportunity__c);
 
-        //query all PSCs associated with these opps, update their ISO code
-        for (Partial_Soft_Credit__c psc : database.query('SELECT Id, Opportunity__c, CurrencyIsoCode FROM Partial_Soft_Credit__c WHERE Opportunity__c IN :oppIds')) {
-            string oppCurrency = mapOppIdToISO.get(psc.Opportunity__c);
+                psc.put('CurrencyIsoCode', oppCurrency);
+                results.add(psc);
+            }
+        } else if (objType == Account_Soft_Credit__c.SObjectType) {
+            //query all AccSCs associated with these opps, update their ISO code
+            for (Account_Soft_Credit__c accountSC : database.query('SELECT Id, Opportunity__c, CurrencyIsoCode FROM Account_Soft_Credit__c WHERE Opportunity__c IN :oppIds')) {
+                string oppCurrency = mapOppIdToISO.get(accountSC.Opportunity__c);
 
-            psc.put('CurrencyIsoCode', oppCurrency);
-            results.add(psc);
+                accountSC.put('CurrencyIsoCode', oppCurrency);
```

Comment:
Seems that the code that updates the AccountSoftCredit and PartialSoftCredit Currency code is not actually working correctly.

In the example below, I started with a $151 USD Opportunity that had one AccountSoftCredit for $151, one PartialSoftCredit for $1 and one Payment for $151. I then changed the Opportunity.Currency to CAD (where the CAD exchange rate is 1.33). When saved, the Opportunity changes to CAD$151 (USD $113.53). The payment update does the same thing (i.e., the two amounts match). That is not the case for the AccountSoftCredit and PartialSoftCredit objects though. In these cases, the amounts changed to CAD$200.83 and CAD$1.33 respectively. See the `PMT_Payment_TDTM.setPaymentCurrencyFromOpportunity` method for an example of how this is handled to ensure the amount is correct. Basically, you have to update both the CurrencyIsoCode *and* the Amount fields. 

![image](https://user-images.githubusercontent.com/801017/43465447-604682fc-94ab-11e8-930c-123f7b3767fa.png)